samples/bluetooth/broadcast_config_tool/src/main.c

```diff
@@ -2365,11 +2476,28 @@ static int cmd_clear(const struct shell *shell, size_t argc, char **argv)
 	return 0;
 }
 
+static void file_paths_get(size_t idx, struct shell_static_entry *entry);
+
+SHELL_DYNAMIC_CMD_CREATE(folder_names, file_paths_get);
+
+static void file_paths_get(size_t idx, struct shell_static_entry *entry)
+{
+	if (idx < num_files_added) {
+		entry->syntax = sd_paths_and_files[idx];
+	} else {
+		entry->syntax = NULL;
+	}
+
+	entry->handler = NULL;
+	entry->help = NULL;
+	entry->subcmd = &folder_names;
+}
+
 SHELL_STATIC_SUBCMD_SET_CREATE(sub_file_cmd,
 			       SHELL_COND_CMD(CONFIG_SHELL, list, NULL, "List files on SD card",
 					      cmd_file_list),
-			       SHELL_COND_CMD(CONFIG_SHELL, select, NULL, "Select file on SD card",
-					      cmd_file_select),
+			       SHELL_CMD_ARG(select, &folder_names, "List files on SD card",
+					     cmd_file_select, 4, 2),
```

Comment:
What does 4 and 2 indicate here? Maybe add a comment since this deviates from the others

Reply:
4 is the number of required arguments and 2 optional arguments. I did have a look at the numbers and it should be 4 and 0.